streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -541,10 +555,33 @@ public void suspend() {
      *                               or if the task producer got fenced (EOS)
      */
     // visible for testing
-    void suspend(final boolean clean) {
+    void suspend(final boolean clean,
+                 final boolean isZombie) {
         closeTopology(); // should we call this only on clean suspend?
         if (clean) {
             commit(false);
+            if (eosEnabled) {
+                recordCollector.close();
+                producer = null;
+            }
+        } else {
+            if (eosEnabled) {
+                try {
+                    if (!isZombie && transactionInFlight) {
+                        producer.abortTransaction();
+                    }
+                    transactionInFlight = false;
```

Comment:
nit: do we want to consider setting `producer` to null here as well if `eosEnabled`?  I realize this branch of the code should only get exercised when closing, but just in case we make changes I don't think it will hurt.